iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/SwitchImpl.java

```diff
@@ -65,7 +65,8 @@ public void setOpen(boolean open) {
         if (oldValue != open) {
             this.open.set(index, open);
             voltageLevel.invalidateCache();
-            network.getListeners().notifyUpdate(this, "open", oldValue, open);
+            String variantId = network.getVariantManager().getVariantId(index);
+            network.getListeners().notifyUpdate(this, "open", variantId, oldValue, open);
```

Comment:
I think we should notify before invalidating the cache of the voltage level (this will send a lot of notifications).